explorer/ast/member_test.cpp

```diff
@@ -0,0 +1,109 @@
+// Part of the Carbon Language project, under the Apache License v2.0 with LLVM
+// Exceptions. See /LICENSE for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include "explorer/ast/member.h"
+
+#include <gmock/gmock.h>
+#include <gtest/gtest.h>
+
+#include <optional>
+
+#include "explorer/ast/bindings.h"
+#include "explorer/ast/declaration.h"
+#include "explorer/ast/expression.h"
+#include "explorer/ast/paren_contents.h"
+#include "explorer/common/arena.h"
+#include "llvm/Support/Casting.h"
+
+// Fake Value classes to avoid depending on /explorer/interpreter
```

Comment:
`interpreter` (as a whole, to generalize), depends on `ast` based on what I saw in the BUILD files. So we probably don't want to introduce a dependency from anything within the `ast` to `interpreter`, to keep things easy to maintain.
And specifically `action_and_value` (and `value.h`) directly includes and depends on `ast`, so I don't believe this would be possible.

I do agree that I don't like that, but because it builds into a separate binary (`expression_test`) which (currently) cannot be linked anywhere else, I think an actual ODR violation is pretty unlikely. I am open to other solutions.

Reply:
`member_test.cpp` should not be part of the `ast` library, so it should be OK for it to depend on both `ast:ast` and `interpreter:action_and_value`. It's not ideal to have a target in the `ast` package depend on one in the `interpreter` package, but it doesn't introduce a cycle between targets, so it shouldn't be disallowed.

Reply:
Ha right, I see what you mean, thank you. Not used to Bazel, I can see it now looking at the `ast` lib definition. Not ideal indeed, but does solve the issue, good with me 👍.